frontend/public/components/silence.tsx

```diff
@@ -156,6 +164,164 @@ const SilencesPage_ = props => <MonitoringListPage
+    const data = Object.assign({}, this.state.data);
+    data.matchers.splice(i, 1);
+    this.setState({data});
+    if (!data.matchers.length) {
+      // All matchers have been removed, so add back a single blank matcher
+      this.addMatcher();
+    }
+  }
+
+  onSubmit = (e): void => {
+    e.preventDefault();
+
+    const alertManagerURL = this.props.urls[MonitoringRoutes.AlertManager];
+    if (!alertManagerURL) {
+      this.setState({error: 'Alertmanager URL not set'});
+      return;
+    }
+
+    this.setState({inProgress: true});
+
+    const body = Object.assign({}, this.state.data);
+    body.startsAt = toISODate(body.startsAt);
+    body.endsAt = toISODate(body.endsAt);
+
+    coFetchJSON.post(`${alertManagerURL}/api/v1/silences`, body)
+      .then(({data}) => {
+        this.setState({error: undefined});
+        history.push(`${SilenceResource.path}/${_.get(data, 'silenceId')}`);
+      })
+      .catch(err => this.setState({error: err.json.error}))
+      .then(() => this.setState({inProgress: false}));
+  }
+  /* eslint-enable no-undef */
+
+  render () {
+    const title = 'Create Silence';
+    const {data, error, inProgress} = this.state;
+
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form className="co-m-pane__body-group silence-form" onSubmit={this.onSubmit}>
+        <SectionHeading text={title} />
+        <p className="co-m-pane__explanation">A silence is configured based on a matcher (label selector). No notification will be sent out for alerts that match all the values or regular expressions.</p>
+        <hr />
+
+        <div className="form-group">
+          <label>Start</label>
+          <Datetime onChange={this.onFieldChange('startsAt')} value={data.startsAt} required />
+        </div>
+        <div className="form-group">
+          <label>End</label>
+          <Datetime onChange={this.onFieldChange('endsAt')} value={data.endsAt} required />
+        </div>
+        <hr />
+
+        <div className="form-group">
+          <label>Matchers</label> (label selectors)
+          <p className="co-help-text">Alerts affected by this silence. Matching alerts must satisfy all of the specified label constraints, though they may have additional labels as well.</p>
+          <div className="row co-m-table-grid__head">
+            <div className="col-xs-4">Name</div>
+            <div className="col-xs-4">Value</div>
+          </div>
+          {_.map(data.matchers, (matcher, i) => <div className="row form-group" key={i}>
+            <div className="col-xs-4">
+              <Text onChange={this.onFieldChange(`matchers[${i}].name`)} placeholder="Name" value={matcher.name} required />
+            </div>
+            <div className="col-xs-4">
+              <Text onChange={this.onFieldChange(`matchers[${i}].value`)} placeholder="Value" value={matcher.value} required />
+            </div>
+            <div className="col-xs-3">
```

Comment:
I think if you use the Bootstrap `checkbox` class here, you could remove the `co-no-bold` below.

```jsx
<div className="col-xs-3 checkbox">
```

Reply:
I tried, but `checkbox` adds unwanted margin, so I left it as `co-no-bold`.